addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -434,52 +442,65 @@ export class HtmlField extends Component {
                 } catch (_e) {
                     return;
                 }
-                cwindow.document
-                    .open("text/html", "replace")
-                    .write(
-                        '<!DOCTYPE html><html>' +
-                        '<head>' +
-                            '<meta charset="utf-8"/>' +
-                            '<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/>\n' +
-                            '<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no"/>\n' +
-                        '</head>\n' +
-                        '<body class="o_in_iframe o_readonly" style="overflow: hidden;">\n' +
-                            '<div id="iframe_target"></div>\n' +
-                        '</body>' +
-                        '</html>');
-
-                for (const cssLib of asset.cssLibs) {
-                    const link = cwindow.document.createElement('link');
-                    link.setAttribute('type', 'text/css');
-                    link.setAttribute('rel', 'stylesheet');
-                    link.setAttribute('href', cssLib);
-                    cwindow.document.head.append(link);
-                }
-                for (const cssContent of asset.cssContents) {
-                    const style = cwindow.document.createElement('style');
-                    style.setAttribute('type', 'text/css');
-                    const textNode = cwindow.document.createTextNode(cssContent);
-                    style.append(textNode);
-                    cwindow.document.head.append(style);
+                if (!this.safeHtmlEdit) {
+                    cwindow.document
+                        .open("text/html", "replace")
+                        .write(
+                            '<!DOCTYPE html><html>' +
+                            '<head>' +
+                                '<meta charset="utf-8"/>' +
+                                '<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/>\n' +
+                                '<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no"/>\n' +
+                            '</head>\n' +
+                            '<body class="o_in_iframe o_readonly" style="overflow: hidden;">\n' +
+                                '<div id="iframe_target"></div>\n' +
+                            '</body>' +
+                            '</html>');
                 }
 
-                const iframeTarget = cwindow.document.querySelector('#iframe_target');
-                iframeTarget.innerHTML = value;
+                if (asset) {
```

Comment:
Unless I'm mistaken it doesn't look like you need the `asset` variable before using it here so I suggest this instead to remove the ternary with `null`:
```js
if (this.props.cssReadonlyAssetId) {
    const asset = await ajax.loadAsset(this.props.cssReadonlyAssetId);
    ... // Use `asset` here.
```